data/avro-schema/src/main/resources/avro/net/corda/data/membership/PersistentMemberInfo.avsc

```diff
@@ -11,13 +11,19 @@
     },
     {
       "name": "memberContext",
-      "doc": "Member provided data in MemberInfo.",
+      "doc": "Deprecated field, signedData should be used instead. Member provided data in MemberInfo.",
       "type": "net.corda.data.KeyValuePairList"
     },
     {
       "name": "mgmContext",
-      "doc": "MGM provided data in MemberInfo.",
+      "doc": "Deprecated field, signedData should be used instead. MGM provided data in MemberInfo.",
       "type": "net.corda.data.KeyValuePairList"
+    },
+    {
```

Comment:
NOTE: AVRO schemas cannot be changed between versions!!

Reply:
When you change an avro schema a new fingerprint is generated for it (a fingerprint is unique to the schema definition in other words). An old version of Corda will already have published all avro schemas to a Kafka topic, keyed by their fingerprint. So if a new version of Corda is picking up an old message, it will be able to grab the old schema by fingerprint from that topic, and the new version of the schema from the compiled JAR, and that's all it needs to maintain backwards compatibility.

So you can change Avro schemas a long as they adhere to the standard Avro rules. I think Avro provide a class to test this: https://avro.apache.org/docs/current/api/java/org/apache/avro/SchemaCompatibility.html
In this case you would be reading messages with the new schema written with the old one.

Obviously your code needs to ensure it can cope with messages all of "versions" whatever that means for any particular message.